Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SANSILLReduction2.py

```diff
@@ -0,0 +1,952 @@
+                             doc='Beam radius [m]; used for beam center search.')
+
+        self.setPropertySettings('BeamRadius', beam)
+
+        self.declareProperty(name='TransmissionBeamRadius',
+                             defaultValue=0.1,
+                             validator=FloatBoundedValidator(lower=0.),
+                             doc='Beam radius [m]; used for flux and transmission calculations.')
+
+        self.setPropertySettings('TransmissionBeamRadius', beam_or_transmission)
+
+        self.declareProperty(name='SampleThickness',
+                             defaultValue=[0.1],
+                             doc='Sample thickness [cm] (if -1, the value is taken from the nexus file).')
+
+        self.setPropertySettings('SampleThickness', water_solvent_sample)
+
+        self.declareProperty(name='WaterCrossSection',
+                             defaultValue=1.,
+                             doc='Provide water cross-section [cm-1]; used only if the absolute scale is performed by dividing to water.')
+
+        self.setPropertySettings('WaterCrossSection', solvent_sample)
+
+        self.declareProperty(name='TransmissionThetaDependent',
+                             defaultValue=True,
+                             doc='Whether or not to use 2theta dependent transmission correction')
+        self.setPropertySettings('TransmissionThetaDependent', can_water_solvent_sample)
+
+        #================================INPUT WORKSPACES================================#
+
+        self.declareProperty(MatrixWorkspaceProperty(name='DarkCurrentWorkspace',
+                                                     defaultValue='',
+                                                     direction=Direction.Input,
+                                                     optional=PropertyMode.Optional),
+                             doc='The name of the Cd/B4C input workspace.')
+
+        self.setPropertySettings('DarkCurrentWorkspace', not_dark)
+
+        self.declareProperty(MatrixWorkspaceProperty(name='EmptyBeamWorkspace',
+                                                     defaultValue='',
+                                                     direction=Direction.Input,
+                                                     optional=PropertyMode.Optional),
+                             doc='The name of the empty beam input workspace.')
+
+        self.setPropertySettings('EmptyBeamWorkspace', not_dark_nor_beam)
+
+        self.declareProperty(MatrixWorkspaceProperty(name='FluxWorkspace',
+                                                     defaultValue='',
+                                                     direction=Direction.Input,
+                                                     optional=PropertyMode.Optional),
+                             doc='The name of the input empty beam flux workspace.')
+
+        self.setPropertySettings('FluxWorkspace', water_solvent_sample)
```

Comment:
In the UI, I see that FluxWorkspace is disabled when ProcessAs=Transmission. Should this be enabled? I was keying in a few of of the calls to SANSILLReduction that were in one of the system tests when I noticed this.